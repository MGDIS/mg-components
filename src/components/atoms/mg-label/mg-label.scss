/**
 * Not scoped into `mg-label` because it could be used outside of an input
 * (ex: All imput with a * are required)
 */
.is-asterisk {
  color: hsl(var(--color-danger));
}
