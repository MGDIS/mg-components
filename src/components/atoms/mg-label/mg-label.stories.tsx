import { h } from "@stencil/core";

export default {
  component: 'mg-label',
  title: 'Atoms/mg-label'
};

const Template = args => {
  // Extract slot so it won't be render as an attribute
  const slot = args.slot;
  delete args.slot;
  // return element
  return <mg-label {...args}>{slot}</mg-label>
};

export const MgLabel = Template.bind({});
MgLabel.args = {
  slot: 'Label',
  identifier: 'identifier',
  required: true,
  colon: true,
};
