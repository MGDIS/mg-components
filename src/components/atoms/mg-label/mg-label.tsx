import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'mg-label',
  styleUrl: 'mg-label.scss',
  shadow: true,
})
export class MgLabel {

  /**
   * Label input id
   */
  @Prop() identifier!: string;

  /**
   * If input is required an asterisk is added at the end of the label
   */
  @Prop() required: boolean;

  /**
   * Add a colon punctuation after label text
   */
  @Prop() colon: boolean;

  /**
   * Switch from label to fieldset sementic
   */
  @Prop() isLegend: boolean = false;

  /*************
   * Lifecycle *
   *************/

  // \u00A0 represent a &nbsp;
  render() {
    const TagName = this.isLegend ? 'legend' : 'label';

    return (
      <TagName class="mg-label" htmlFor={this.identifier}>
        <slot></slot>
        { this.required && [`\u00A0`, <span class="is-asterisk">*</span>] }
        { this.colon && `\u00A0:` }
      </TagName>
    );
  }

}
