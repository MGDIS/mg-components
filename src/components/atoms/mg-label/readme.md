This atom refers to thes [PDA9-64](https://jira.mgdis.fr/browse/PDA9-64), [PDA9-723](https://jira.mgdis.fr/browse/PDA9-723).

<!-- Auto Generated Below -->


## Properties

| Property                  | Attribute    | Description                                                       | Type      | Default     |
| ------------------------- | ------------ | ----------------------------------------------------------------- | --------- | ----------- |
| `colon`                   | `colon`      | Add a colon punctuation after label text                          | `boolean` | `undefined` |
| `identifier` _(required)_ | `identifier` | Label input id                                                    | `string`  | `undefined` |
| `required`                | `required`   | If input is required an asterisk is added at the end of the label | `boolean` | `undefined` |


## Dependencies

### Used by

 - [mg-input-checkbox](../../molecules/inputs/mg-input-checkbox)
 - [mg-input-date](../../molecules/inputs/mg-input-date)
 - [mg-input-numeric](../../molecules/inputs/mg-input-numeric)
 - [mg-input-select](../../molecules/inputs/mg-input-select)
 - [mg-input-text](../../molecules/inputs/mg-input-text)
 - [mg-input-textarea](../../molecules/inputs/mg-input-textarea)

### Graph
```mermaid
graph TD;
  mg-input-checkbox --> mg-label
  mg-input-date --> mg-label
  mg-input-numeric --> mg-label
  mg-input-select --> mg-label
  mg-input-text --> mg-label
  mg-input-textarea --> mg-label
  style mg-label fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
