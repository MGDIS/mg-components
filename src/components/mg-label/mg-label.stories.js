export default {
  component: 'mg-label',
  title: 'Atoms/mg-label',
  parameters: {
    docs: {
      source: {
        code: `<mg-label required colon>Label</mg-label>`,
      },
    },
  },
};

const Template = args => <mg-label {...args}>{args.slot}</mg-label>;

export const MgLabel = Template.bind({});
MgLabel.args = {
  slot: 'Label',
  reference: 'reference-id',
  required: true,
  colon: true,
};
