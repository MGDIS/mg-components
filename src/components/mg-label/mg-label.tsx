import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'mg-label',
  styleUrl: 'mg-label.scss',
  scoped: true,
})
export class MgLabel {

  /**
   * Internal
   */
  private classes = ['mg-label'];

  /**
   * Label input reference
   * To match for/id
   */
  @Prop() reference: string;

  /**
   * If input is required an asterisk is added at the end of the label
   */
  @Prop() required: boolean;

  /**
   * Add a colon punctuation after label text
   */
   @Prop() colon: boolean;

  /**
   * Render
   */
  render() {
    return (
      <label class={this.classes.join(' ')} htmlFor={this.reference}>
        <slot></slot>
        { this.required && <span aria-hidden="true">&nbsp;*</span> }
        { this.colon && `\u00A0:` /* represent a &nbsp; */ }
      </label>
    );
  }

}
