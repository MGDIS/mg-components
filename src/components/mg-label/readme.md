# mg-label



<!-- Auto Generated Below -->


## Properties

| Property    | Attribute   | Description                                                       | Type      | Default     |
| ----------- | ----------- | ----------------------------------------------------------------- | --------- | ----------- |
| `colon`     | `colon`     | Add a colon punctuation after label text                          | `boolean` | `undefined` |
| `reference` | `reference` | Label input reference To match for/id                             | `string`  | `undefined` |
| `required`  | `required`  | If input is required an asterisk is added at the end of the label | `boolean` | `undefined` |


## Dependencies

### Used by

 - [mg-input-text](../mg-input-text)

### Graph
```mermaid
graph TD;
  mg-input-text --> mg-label
  style mg-label fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
