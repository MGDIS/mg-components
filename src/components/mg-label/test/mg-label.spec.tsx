import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { MgLabel } from '../mg-label';

const getPage = (args) => newSpecPage({
  components: [MgLabel],
  template: () => (<mg-label {...args}>mg-label</mg-label>),
});

describe('mg-label', () => {
  test.each([false, true])('Should render label with required %s', async (required) => {
    const args = {reference: 'blu', required};
    const { root } = await getPage(args);

    expect(root).toEqualHtml(`
      <mg-label>
        <label htmlfor="${args.reference}">
          mg-label
          ${ required ? '<span aria-hidden="true"> *</span>' : ''}
        </label>
      </mg-label>
    `);
  });
});
