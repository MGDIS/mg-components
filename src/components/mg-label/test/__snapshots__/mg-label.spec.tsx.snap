// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mg-label Should render label with required false and colon false 1`] = `
<mg-label>
  <!---->
  <label class="mg-label">
    mg-label
  </label>
</mg-label>
`;

exports[`mg-label Should render label with required false and colon true 1`] = `
<mg-label>
  <!---->
  <label class="mg-label">
    mg-label
    :
  </label>
</mg-label>
`;

exports[`mg-label Should render label with required true and colon false 1`] = `
<mg-label>
  <!---->
  <label class="mg-label">
    mg-label
    <span aria-hidden="true">
      *
    </span>
  </label>
</mg-label>
`;

exports[`mg-label Should render label with required true and colon true 1`] = `
<mg-label>
  <!---->
  <label class="mg-label">
    mg-label
    <span aria-hidden="true">
      *
    </span>
    :
  </label>
</mg-label>
`;
